pkg/specs/pvcs.go

```diff
@@ -176,12 +177,55 @@ func DetectPVCs(
 		result.Dangling = append(result.Dangling, pvc.Name)
 	}
 
+	if !cluster.ShouldCreateWalArchiveVolume() {
+		return result
+	}
+
+	for _, instance := range getInstancesPods(podList) {
+		expectedPVCs := getExpectedInstancePVCNames(cluster, instance.Name)
+		var indexOfFoundPVCs []int
+
+		for idx, pvcName := range result.Healthy {
+			if slices.Contains(expectedPVCs, pvcName) {
+				indexOfFoundPVCs = append(indexOfFoundPVCs, idx)
+			}
+		}
+
+		switch {
+		case len(expectedPVCs) == len(indexOfFoundPVCs):
+			continue
```

Comment:
hmm, we continue in all the cases, no? Then this can be expressed more explicitly

Reply:
Not sure that I got the question correctly, but it can happen that one PVC is gone and then we need to null the other too, see the  case `len(expectedPVCs) > len(indexOfFoundPVCs):`

Once we have more than one volume (not just the PGDATA) all of them are needed to make the pod start correctly